utils/src/sequence/mod.rs

```diff
@@ -42,8 +42,6 @@ pub trait Span:
     + Debug
     + Hash
     + Display
-    + AsRef<[u8]>
```

Comment:
In practice, not requiring `Span` to readily provide a `[u8]` representation makes it much easier to work with (particularly enums because we must serve an `as_ref()` -> requires a struct).